core/src/test/java/org/wso2/msf4j/interceptor/TestInterceptorDeprecated.java

```diff
@@ -0,0 +1,72 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.msf4j.interceptor;
+
+import org.wso2.msf4j.Interceptor;
+import org.wso2.msf4j.Request;
+import org.wso2.msf4j.Response;
+import org.wso2.msf4j.ServiceMethodInfo;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class TestInterceptorDeprecated implements Interceptor {
```

Comment:
Comment added explaining the purpose of the class. This is a class written using the previous "interceptors" (deprecated) to check whether the previous are working as expected. Once the deprecated interceptors are removed from the MSF4J this interceptor along with the relevant tests will be removed